File: Data Structure/20255/Solutions/Find_Maximum_Element_in_Array.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution{
public:
    int findMaximum(vector<int>& arr){
        if(arr.empty()) return 0;
        int maxVal = arr[0];
        for(int i = 1; i < arr.size(); i++){
            if(arr[i] > maxVal) maxVal = arr[i];
        }
        return maxVal;
    }
};
int main(){
    int n; cin >> n;
    if(n == 0){
        cout << "Array is Empty";
        return 0;
    }
    vector<int> arr(n);
    for(int i = 0; i < n; i++) cin >> arr[i];
    Solution sol;
    int ans = sol.findMaximum(arr);
    cout << ans;
    return 0;
}

File: Data Structure/20255/Solutions/Find_Maximum_Element_in_Array.java
import java.util.*;
class Solution{
    Integer findMaximum(int[] arr){
        if(arr == null || arr.length == 0) return null;
        int max = arr[0];
        for(int i = 1; i < arr.length; i++){
            if(arr[i] > max) max = arr[i];
        }
        return max;
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; i++) arr[i] = sc.nextInt();
        Solution sol = new Solution();
        Integer ans = sol.findMaximum(arr);
        if(ans == null) System.out.println("Array is Empty");
        else System.out.println(ans);
    }
}

